app/view/components/StyledText.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import {I18nManager} from 'react-native';
+
+import {useTheme, Text} from 'react-native-paper';
+
+const StyledText = (props) => {
+  const {colors, fonts} = useTheme();
+  const {textColor = colors.text, family = 'regular', style, ...rest} = props;
+
+  const font = fonts[family];
+  const writingDirection = I18nManager.isRTL ? 'rtl' : 'ltr';
+
+  return (
+    <Text
+      {...rest}
+      style={[
+        {color: textColor, ...font, textAlign: 'left', writingDirection},
+        style,
```

Comment:
You use an array to combine styles, nice, but assume the `style` that comes from props is a combined array. how you wanna settle this issue?

Reply:
I think Style prop is working with dimensional array 
like 
` style={[
           [{color: 'red'},{fontSize:50}],
          {backgroundColor: 'blue'}
  ]}`

Reply:
I don't think so unless after version 0.59 this part has serious changes. when I pass some deep dimensional arrays it fell into error. something like that:

```javascript
style={[ [ [ [{color: 'red'},{fontSize:50}], {backgroundColor: 'blue'} ], {backgroundColor: 'blue'} ], {backgroundColor: 'blue'} ]}
```
If you are sure about it. please _Resolve_ this conversation. but if not please examine it.